documentation/src/test/java/org/junit/api/tools/ApiReportGenerator.java

```diff
@@ -67,43 +67,45 @@ ApiReport generateReport(String... packages) {
 		final String EOL = System.lineSeparator();
 
 		// Scan packages
-		ScanResult scanResult = new ClassGraph().whitelistPackages(packages).enableAnnotationInfo().scan();
-
-		// Collect names
-		ClassInfoList classesWithApiAnnotation = scanResult.getClassesWithAnnotation(API.class.getCanonicalName());
-		List<String> names = classesWithApiAnnotation.getNames();
-
-		logger.debug(() -> {
-			StringBuilder builder = new StringBuilder(
-				names.size() + " @API declarations (including meta) found in class-path:");
-			builder.append(EOL);
-			scanResult.getClasspathURLs().forEach(e -> builder.append(e).append(EOL));
-			return builder.toString();
-
-		});
-
-		// Collect types
-		List<Class<?>> types = classesWithApiAnnotation.loadClasses();
-		// only retain directly annotated types
-		types.removeIf(c -> !c.isAnnotationPresent(API.class));
-		types.sort(Comparator.comparing(Class::getName));
-
-		logger.debug(() -> {
-			StringBuilder builder = new StringBuilder("Listing of all " + types.size() + " annotated types:");
-			builder.append(EOL);
-			types.forEach(e -> builder.append(e).append(EOL));
-			return builder.toString();
-		});
-
-		// Build map
-		Map<Status, List<Class<?>>> declarationsMap = new EnumMap<>(Status.class);
-		for (Status status : Status.values()) {
-			declarationsMap.put(status, new ArrayList<>());
+		try (ScanResult scanResult = new ClassGraph().whitelistPackages(
```

Comment:
I guess, it should be enough to wrap the `new ClassGraph().whitelistPackages(packages).enableAnnotationInfo().scan()` call in a `try-with-resources` block. If that's the case,  the `scanResult` should be declared before the block and used after it. Thus, no need to indent the next 30 lines.

Reply:
Your explanatio and @lukehutch 's comment convinced me.